GeneratorInterface/SherpaInterface/src/SherpaHadronizer.cc

```diff
@@ -186,17 +188,21 @@ SherpaHadronizer::SherpaHadronizer(const edm::ParameterSet &params) :
   std::string shRng  = "EXTERNAL_RNG=CMS_SHERPA_RNG";
 
   //create the command line
-  arguments.push_back(shRun.c_str());
-  arguments.push_back(shPath.c_str());
-  arguments.push_back(shPathPiece.c_str());
-  arguments.push_back(shRes.c_str());
-  arguments.push_back(shRng.c_str());
+  arguments.push_back(shRun);
+  arguments.push_back(shPath);
+  arguments.push_back(shPathPiece);
+  arguments.push_back(shRes);
+  arguments.push_back(shRng);
   isInitialized=false;
  //initialization of Sherpa moved to initializeForInternalPartons
 }
 
 SherpaHadronizer::~SherpaHadronizer()
 {
+  Generator->~Sherpa();
+  #ifdef USING__MPI
+    MPI::Finalize();
```

Comment:
so this is a sign that you can not use the default CMSSW builds.. have you considered run time switches to provide this same functionality?

Reply:
sherpa is either compiled with or without mpi support, so runtime switches are not an option
if it is compiled with mpi support it needs the mpi init and finalize calls even if it is not parallelized using mpi (this is the case here)